extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisRecordSupplier.java

```diff
@@ -296,14 +279,22 @@ private Runnable fetchRecords()
             );
 
 
-            log.trace(
-                "Stream[%s] / partition[%s] / sequenceNum[%s] / bufferRemainingCapacity[%d]: %s",
-                currRecord.getStream(),
-                currRecord.getPartitionId(),
-                currRecord.getSequenceNumber(),
-                records.remainingCapacity(),
-                currRecord.getData().stream().map(StringUtils::fromUtf8).collect(Collectors.toList())
```

Comment:
I'm not even sure this is safe to do. Some byte arrays could technically be invalid UTF-8 sequences, which would lead to exceptions when trying to log values.

Reply:
It's OK, because `fromUtf8` doesn't throw exceptions, it just replaces invalid sequences with the [replacement character](https://www.fileformat.info/info/unicode/char/fffd/index.htm). Although the log messages will look very weird.

Reply:
good point